backend/resources/contact.py

```diff
@@ -1,34 +1,77 @@
-import os
+import logging
+from textwrap import dedent
 
 from http import HTTPStatus
 
 from flask import request
-from flask_mail import Message
+from flask import current_app
 from flask_restful import Resource
 
 from marshmallow import ValidationError
 
 from backend.serializers.message_serializer import MessageSchema
 from backend.extensions import mail
+from backend.helpers import wrap_io
+
+
+MAIL_RECIPIENT = current_app.config["MAIL_RECIPIENT"]
+MAIL_WEB_SERVER = current_app.config["MAIL_WEB_SERVER"]
+BASE_URL = current_app.config["BASE_URL"]
 
 
 class SendMessage(Resource):
     def post(self):
         try:
             message_schema = MessageSchema()
-            new_msg = message_schema.load(request.json)
+            data = message_schema.load(request.json)
         except ValidationError as err:
             return {"message": err.messages}, HTTPStatus.BAD_REQUEST
-        name, phone, content = new_msg["name"], new_msg["phone"], new_msg["content"]
-        base_url = os.environ.get("BASE_URL", "codeforpoznan.pl")
-
-        msg = Message(
-            subject=f"Email z {base_url} od {name}",
-            sender=new_msg["email"],
-            reply_to=new_msg["email"],
-            recipients=[f"hello@{base_url}"],
+
+        name, email = data["name"], data["email"]
+        phone, content = data["phone"], data["content"]
+
+        subject = f"""
+            Email z {BASE_URL} od {name}
+        """
+
+        body = f"""
+            Nowa wiadomość od {name}
+            numer telefonu: {phone}
+            Treść:
+            {content}
+        """
+
+        status_code = self._send_message(
+            sender=email,
+            reply_to=email,
+            recipients=[MAIL_RECIPIENT],
+            subject=dedent(subject).strip(),
+            body=dedent(body).strip(),
         )
-        msg.body = f"Nowa wiadomość od {name}, nr tel: {phone} \nTreść:\n {content}"
-        mail.send(msg)
 
-        return {"message": "Contact message successfully sent"}, HTTPStatus.OK
+        if status_code == HTTPStatus.OK:
+            return {"message": "Contact message successfully sent"}, status_code
+        else:
+            return {"message": "Failed to send contact message"}, status_code
+
+    @staticmethod
+    def _send_message(**kwargs):
+        with wrap_io() as (out, err):
+            mail.send_message(**kwargs)
+
+        # known mail testing domain, find message link in output
+        if MAIL_WEB_SERVER:
+            if "MSGID=" in err():
+                msgid = err().split("MSGID=")[1].split("]")[0]
```

Comment:
ethereal mail sends unique message id in SMTP response which we want to parse and print, because it's the key to the message link (see commands below).